core/network/impl/gossiper_broadcast.cpp

```diff
@@ -17,7 +17,75 @@ namespace kagome::network {
   void GossiperBroadcast::reserveStream(
       const libp2p::peer::PeerInfo &peer_info,
       std::shared_ptr<libp2p::connection::Stream> stream) {
-    streams_.emplace(peer_info, std::move(stream));
+    reserved_streams_.emplace(peer_info, std::move(stream));
+  }
+
+  void GossiperBroadcast::addStream(
+      std::shared_ptr<libp2p::connection::Stream> stream) {
+    auto peer_id_res = stream->remotePeerId();
+    if (not peer_id_res.has_value()) {
+      logger_->error("Can't get peer_id: {}", peer_id_res.error().message());
+      return stream->reset();
+    }
+    auto &peer_id = peer_id_res.value();
+
+    auto syncing_stream_it = syncing_streams_.find(peer_id);
+    if (syncing_stream_it != syncing_streams_.end()) {
+      if (syncing_stream_it->second != stream) {
+        syncing_stream_it->second->reset();
+        syncing_stream_it->second.swap(stream);
+        logger_->debug("Syncing stream (peer_id={}) was replaced",
+                       peer_id.toHex());
+      }
+      return;
+    }
+
+    logger_->debug("Syncing stream (peer_id={}) was emplaced", peer_id.toHex());
+  }
```

Comment:
It seems we miss insertion added stream to ```syncing_streams_``` here.

Reply:
I don't see an insertion place anywhere else.

Reply:
I tried, but it breaks tworking. Gossiper must be refactored before: without protocol_id streams replaces at any collision by peer_id.